cachix-api/src/Cachix/Api.hs

```diff
@@ -106,13 +109,21 @@ data BinaryCacheStreamingAPI route
           :: route
                :- CachixAuth
                :> "nar"
-               :> Capture "nar" NarC
+               :> Capture "nar" NarFileName
                :> StreamGet NoFraming OctetStream (ConduitT () ByteString (ResourceT IO) ()),
         createNar
           :: route
                :- "nar"
                :> StreamBody NoFraming XNixNar (ConduitT () ByteString (ResourceT IO) ())
-               :> Post '[JSON] NoContent
+               :> Post '[JSON] NoContent,
+        serveNar 
+          :: route
+              :- CachixAuth
+              :> "serve"
```

Comment:
Idea:

```suggestion
              :> "nix"
              :> "store"
```

Reply:
I'm not  sure this is a good idea. If we ever allow changing `/nix/store` prefix, this will look weird.

It's also not entirely full path, since we're using store hash and not store path.

Reply:
Just need to strip the `-$name` part to make it work based on hash, or even better is to query the real store path. It's in the narinfo.
If we allow changing the prefix, we can do a `CaptureAll` at the root and make sure it's the last route that is tried.
For security we can whitelist or blacklist prefixes. Limiting to `/nix/store` is a good start.
Serving broken links is bad ux.

Reply:
This should be done separately. Resolving.